Move the credential chooser to after the popup/redirect (#29)
fedidcg/CrossSiteCookieAccessCredential: README.md

## Key scenarios

These APIs together enable login and linking scenarios that I have put into a few categories.
For all of these, imagine that an identity provider would store a credential on the user's browser when they log in.

### Scenario 1: User intends to link to an identity provider they are not logged into

In this case, our user is not even logged into this this identity provider (`idp.net`), just having navigated to this site (`example.com`).
They first interact with some UI in the page that is clearly associated with the identity provider and the following is called.

```js
let credential = await navigator.credentials.get({
  identity : {
    providers : [
      {
        loginURL : "https://login.idp.net/login.html",
        origin: "https://login.idp.net", // may be omitted, inferred from loginURL
      },
    ]
  }
});
```

The browser sees there is no credential in the credential store that would work for `example.com`.

> So it falls back and opens the `loginURL` in a tab of the RP's choosing.

Reviewer: ```suggestion
So it falls back and opens the `loginURL`. The RP can choose whether to open this URL in a pop-up or via a redirect.
```

> Since it is not specified, we assume the current tab.

Reviewer: ```suggestion
The API defaults to redirecting the user to the `loginURL`.
```

There, the user may do some auth flow and on completion, the identity provider calls the following:
The browser sees there is no credential in the credential store that would work for `example.com`.
So it falls back and opens the `loginURL` in a tab of the RP's choosing.
Since it is not specified, we assume the current tab.
There, the user goes through some authentication and/or authorization flow entirely of the identity provider's choosing, after which the identity provider stores a credential with some code like this:

```js
let cred = await navigator.credentials.create({
  identity : {
    effectiveQueryURL: "https://api.login.idp.net/v1/foo",
    uiHint: {
      name: "example human readable",
      iconURL: "https://api.login.idp.net/v1/photos/exampleUser",

>     }

Reviewer: Why no UI hints? 😨

Author: They still exist lower down. Just trimming up the example to the core functional bits.

      {
        origin: "https://login.idp.net",
      },
    ]
  }
});
```

> We need to bypass the requirement for user interaction here if we had user interaction to force the navigation but redirected the current tab instead of opening a popup.

Reviewer: This seems a bit involved so it might make sense to track / write up this requirement and how it would work in a separate issue vs. just mentioning it here. So I would just remove this and file a new issue instead.

Note that in the case of a popup, the credential chooser should show once the identity provider stores a credential that is effective for the pending credential request on the relying party, removing the need for the relying party to call `navigator.credentials.get` a second time.

> ### Scenario 2: User logs in with one of many identity providers, or other types of credential

Reviewer: I'm not 100% sure what Scenario 2 adds to the proposal. What is it that we want user agents to implement here?

The specifics of that interaction dictate what Credential types are appropriate.
For sake of discussion, let's say the identity providers defined here and a PasswordCredential would be good.
The page then calls the following:

```js
let credential = await navigator.credentials.get({
  password: true,
  identity : {
    providers : [
      {
        origin: "https://login.idp.net",
      },
      // ... many allowed ...
      {
        origin: "https://auth.example.biz",
      },
    ]
  }
});
```

Then the user is presented any account information from identity providers they have visited and stored themselves in the credential store, and password manager entries as options in the browser UI.
Whichever is selected is returned.

Note also that depending on the credential manager state, request details, and if only one credential is collected from the store, the UI may be elided.
Or if the browser simply wants to poll for the presence of such a credential without any UI they can do that as well.
See the [mediation requirements in the Credential Manager API](https://w3c.github.io/webappsec-credential-management/#mediation-requirements) for more details.

> ### Scenario 3: User intends to link to an identity provider they are already logged in to

Reviewer: Same with Scenario 3, what is the delta to FedCM?

    providers : [
      {
        origin: "https://login.idp.net",
        loginURL: "https://bounce.example.com/?u=https://login.idp.net/login.html?r=https://rp.net/",
        loginTarget: "redirect",
      },
    ]
  }
});
```

> This example shows the use perfect for a "Log in with Foo" button, where one identity provider is presented, and if the user has not already logged in, they may be redirected to that provider's login page. This redirect behavior is only permitted when there is only one provider in the list. A provider with `loginURL` field indicates that this is the expected mode. If `loginURL` is present, but `origin` is not, its value can be inferred as the origin of the link.

Reviewer: What's "the use perfect"?

Author: it could be rewritten as "the use of the API that is perfect". "use" as in the noun meaning occasion or need. Agreed that this could be reworded

Reviewer 2: ```suggestion
This example shows a scenario that is perfect for a "Log in with Foo" button, where one identity provider is presented, and if the user has not already logged in, they may be redirected to that provider's login page. This redirect behavior is only permitted when there is only one provider in the list. A provider with `loginURL` field indicates that this is the expected mode. If `loginURL` is present, but `origin` is not, its value can be inferred as the origin of the link.
```
